packages/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -354,11 +349,7 @@ private function isObjectTypeOfObjectType(ObjectType $resolvedObjectType, Object
         }
 
         $classReflection = $this->reflectionProvider->getClass($resolvedObjectType->getClassName());
-        if (! isset($this->traitExistsCache[$classReflection->getName()])) {
```

Comment:
I think since we already checked with `hasClass` that `$resolvedObjectType->getClassName()` exists, we only need to check with `isTrait()` whether its a trait.

in my thinking it should be equivalent to what this cache was doing before.

@keulinho could you verify this has the same perf characteristics for your use case, as measured in https://github.com/rectorphp/rector-src/pull/3501 ?

Reply:
The change in general looks good to me and the solution way more elegant that mine :see_no_evil: 

actually i've paused trying to get rector running smoothly with our codebase for now, i will get back when i have the time to look at it again